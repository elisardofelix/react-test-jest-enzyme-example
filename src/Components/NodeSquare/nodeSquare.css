.node-square {
	background: #ffffff;
	border-radius: 3px;
	width: 648px;
	filter: drop-shadow(0px 4px 8px rgba(84, 110, 122, 0.24));
	padding: 14px 12px 16px;
	margin-bottom: 12px;
}

.node-square-header {
	display: flex;
	justify-content: space-between;
	align-content: center;
}

.node-square-title {
	font-family: Roboto;
	font-style: normal;
	font-weight: normal;
	font-size: 16px;
	line-height: 24px;
	margin: 0;
	/* or 150% */

	letter-spacing: 0.44px;

	/* Dark Color */

	color: #263238;
}

.node-square-isactive {
	font-family: Roboto;
	font-style: normal;
	font-weight: 500;
	font-size: 10px;
	line-height: 16px;
	/* or 160% */

	display: flex;
	align-items: center;
	letter-spacing: 1.5px;
	text-transform: uppercase;

	/* Dark Color */

	color: #263238;
}

.node-square-active::before {
	content: "";
	display: inline-block;
	width: 6px;
	height: 6px;
	border-radius: 7.5px;
	background-color: #18cc55;
	transform: translateY(-1px);
	margin-right: 5px;
}

.node-square-inactive::before {
	content: "";
	display: inline-block;
	width: 6px;
	height: 6px;
	border-radius: 7.5px;
	background-color: #eb5757;
	transform: translateY(-1px);
	margin-right: 5px;
}

.node-square-url {
	/* 14 sp â€¢ Body 2 */

	font-family: Roboto;
	font-style: normal;
	font-weight: normal;
	font-size: 14px;
	line-height: 20px;
	/* identical to box height, or 143% */

	display: flex;
	align-items: center;
	letter-spacing: 0.25px;

	/* Dark Color */

	color: #263238;
	opacity: 0.54;
}

.node-square-arrow {
	margin: 0 0 0 4px;
	cursor: pointer;
}

.node-square-arrow-up {
	transform: rotate(180deg);
}
