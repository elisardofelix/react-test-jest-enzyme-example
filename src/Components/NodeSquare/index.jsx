import React, { useState } from "react";
import NodeMessage from "../NodeMessage";
import "./nodeSquare.css";
import Arrow from "../../Assets/down-arrow.svg";

const NodeSquare = ({ title, url, isActive, messages }) => {
	const [showMessages, setShowMessages] = useState(false);
	const arrowHandler = () => {
		setShowMessages(!showMessages);
	};
	return (
		<div className="node-square">
			<div className="node-square-header">
				<h2 className="node-square-title">{title}</h2>{" "}
				<span
					className={`node-square-isactive ${
						isActive ? "node-square-active" : "node-square-inactive"
					}`}
				>
					{isActive ? "ONLINE" : "OFFLINE"}
					<img
						className={`node-square-arrow ${
							showMessages && "node-square-arrow-up"
						}`}
						src={Arrow}
						alt=""
						onClick={arrowHandler}
					/>
				</span>
			</div>

			<span className="node-square-url">{url}</span>

			{showMessages
				? messages.map((item, key) => <NodeMessage key={key} {...item} />)
				: null}
		</div>
	);
};

export default NodeSquare;
