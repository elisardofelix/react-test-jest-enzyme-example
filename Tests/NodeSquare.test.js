import React from "react";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";
import NodeSquare from "../src/Components/NodeSquare";

const initialData = {
	title: "Node 01",
	url: "noseque.herokuapp.com/1237asj",
	isActive: true,
	messages: [
		{
			id: 1,
			message: "un mensaje de prueba 01",
		},
		{
			id: 2,
			message: "un mensaje de prueba 01",
		},
		{
			id: 3,
			message: "un mensaje de prueba 01",
		},
		{
			id: 4,
			message: "un mensaje de prueba 01",
		},
		{
			id: 5,
			message: "un mensaje de prueba 01",
		},
		{
			id: 6,
			message: "un mensaje de prueba 01",
		},
		{
			id: 7,
			message: "un mensaje de prueba 01",
		},
		{
			id: 8,
			message: "un mensaje de prueba 01",
		},
	],
};

describe("NodeSquare component", () => {
	it("should render snapshot", () => {
		const component = renderer.create(<NodeSquare {...initialData} />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("should render required node elements", () => {
		const Node = shallow(<NodeSquare {...initialData} />);

		const title = Node.find({ className: "node-square-title" });
		expect(title.text()).toBe("Node 01");
	});

	it("should display the messages on click and hide on click again arrow", () => {
		const Node = mount(<NodeSquare {...initialData} />);

		let messages = Node.find({ className: "node-message" });

		expect(messages.length).toBe(0);

		Node.find("img.node-square-arrow").simulate("click");

		messages = Node.find({ className: "node-message" });

		expect(messages.length).toBe(8);

		Node.find("img.node-square-arrow").simulate("click");

		messages = Node.find({ className: "node-message" });

		expect(messages.length).toBe(0);
	});
});
