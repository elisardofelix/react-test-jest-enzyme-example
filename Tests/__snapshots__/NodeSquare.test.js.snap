// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeSquare component should render snapshot 1`] = `
<div
  className="node-square"
>
  <div
    className="node-square-header"
  >
    <h2
      className="node-square-title"
    >
      Node 01
    </h2>
     
    <span
      className="node-square-isactive node-square-active"
    >
      ONLINE
      <img
        alt=""
        className="node-square-arrow false"
        onClick={[Function]}
        src="test-file-stub"
      />
    </span>
  </div>
  <span
    className="node-square-url"
  >
    noseque.herokuapp.com/1237asj
  </span>
</div>
`;
